integration_tests/commands/async/deque_test.go

```diff
@@ -442,6 +442,40 @@ func TestLLEN(t *testing.T) {
 	deqCleanUp(conn, "k")
 }
 
+func TestLInsert(t *testing.T) {
+	deqTestInit()
+	conn := getLocalConnection()
+	defer conn.Close()
+
+	testCases := []struct {
+		name   string
+		cmds   []string
+		expect []any
+	}{
+		{
+			name:   "LINSERT before",
+			cmds:   []string{"LPUSH k v1 v2 v3 v4", "LINSERT k before v2 e1", "LINSERT k before v1 e2", "LINSERT k before v4 e3"},
```

Comment:
The test case should also verify the final state of the list by running `LRANGE` at the end.

Reply:
We don't have LRANGE command implemented yet. I didn't want to add it as a part of this PR since it will get too long. It's pretty straight forward to implement LRANGE after the introduction of the iterator in this PR. Let's create a new issue request for this and update these tests then. For now I can add a TODO item here. 

Let me know your thoughts here.

Reply:
I see, that sounds good. Please add a todo, additionally, it would be good to create an issue as well for the same to allow for better tracking of work. You can assign the issue to yourself.

Reply:
On second thought, maybe it is better to implement LRANGE as a part of this PR itself? It will allow us to verify functionality for this command as well. What do you think?

Reply:
Sure, I can add implementation for LRANGE as well. My only concern was size of the PR, but that should be okay it seems.